source/binding/AssignmentExpressions.cpp

```diff
@@ -105,7 +105,8 @@ namespace slang {
 Expression& Expression::implicitConversion(const BindContext& context, const Type& targetType,
                                            Expression& expr) {
     ASSERT(targetType.isAssignmentCompatible(*expr.type) ||
-           (targetType.isString() && expr.isImplicitString()) ||
+           (expr.isImplicitString() &&
+            (targetType.isString() || targetType.isUnpackedArrayOfByte())) ||
```

Comment:
Could simplify this a bit by using the Type::canBeStringLike() method here.

Reply:
canBeStringLike() accepts dynamic arrays. As mentioned above, for safety I will limit the enhancement to fixed-size unpacked array for now, so canBeStringLike() does not look good for now.